ums-core/src/main/java/org/ums/persistent/dao/PersistentApplicationTESDao.java

```diff
@@ -0,0 +1,749 @@
+    String query = getCoursesForMapping;
+    return mJdbcTemplate.query(query, new Object[] {pFacultyId, pSemesterId},
+        new ApplicationTESRowMapperForAssignedCoursesForMapping());
+  }
+
+  @Override
+  public List<MutableApplicationTES> getAssignedCourses(String pFacultyId, Integer pSemesterId) {
+    String query = getAssignedCourses;
+    return mJdbcTemplate.query(query, new Object[] {pFacultyId, pSemesterId},
+        new ApplicationTESRowMapperForAssignedCourses());
+  }
+
+  @Override
+  public List<ApplicationTES> getFacultyMembers(String pDeptId) {
+    String query = getFacultyMembers;
+    return mJdbcTemplate.query(query, new Object[] {pDeptId}, new ApplicationTESRowMapperForFacultyMembers());
+  }
+
+  @Override
+  public List<ApplicationTES> getAlreadyReviewdCourses(String pStudentId, Integer pSemesterId) {
+    String query = ALREADY_REVIEWED;
+    return mJdbcTemplate.query(query, new Object[] {pStudentId, pSemesterId},
+        new ApplicationTESRowMapperForGetAlreadyReviewedCourses());
+  }
+
+  @Override
+  public Long addQuestions(MutableApplicationTES pMutableList) {
+    String query = addQuestion;
+    String getMax = "SELECT max(QUESTION_ID) from APPLICATION_TES_QUESTIONS";
+    Integer questionId = mJdbcTemplate.queryForObject(getMax, Integer.class);
+    questionId = questionId + 1;
+
+    Long id = mIdGenerator.getNumericId();
+    mJdbcTemplate.update(query, id, questionId, pMutableList.getQuestionDetails(), pMutableList.getSemester(),
+        pMutableList.getObservationType());
+    return id;
+  }
+
+  @Override
+  public List<Long> saveAssignedCourses(List<MutableApplicationTES> pMutableList) {
+    List<Object[]>  parameters  =  getInsertParamListForHeadTes(pMutableList);
+    mJdbcTemplate.batchUpdate(INSERT_ASSIGNED_COURSES,  parameters);
+    return  parameters.stream()
+            .map(paramArray  ->  (Long)  paramArray[0])
+            .collect(Collectors.toCollection(ArrayList::new));
+  }
+
+  @Override
+  public List<Long> setQuestions(List<MutableApplicationTES> pMutableList) {
+    List<Object[]>  parameters  =  getInsertParamListForSetQuestions(pMutableList);
+    mJdbcTemplate.batchUpdate(setQuestion,  parameters);
+    return  parameters.stream()
+            .map(paramArray  ->  (Long)  paramArray[0])
+            .collect(Collectors.toCollection(ArrayList::new));
+  }
+
+  @Override
+  public  List<Long>  create(List<MutableApplicationTES>  pMutableList)  {
+    List<Object[]>  parameters  =  getInsertParamList(pMutableList);
+    mJdbcTemplate.batchUpdate(INSERT_ONE,  parameters);
+    return  parameters.stream()
+            .map(paramArray  ->  (Long)  paramArray[0])
+            .collect(Collectors.toCollection(ArrayList::new));
+  }
+
+  @Override
+  public int delete(List<MutableApplicationTES> pMutableList) {
+    String query = DELETE_ALL + " WHERE QUESTION_ID=? AND SEMESTER_ID=?";
+    List<Object[]> parameters = deleteParamList(pMutableList);
```

Comment:
Why do we need semestr_id here? Isn't ID enough here?

Reply:
eta tes_Set_Question Table er
Ekhen eki Questionid  fall17,Spring 17 e set hoyte pare
so only question id diye delet korle sob delet hoye jabe